### DISCUSSION TITLE
Whitespace

### DISCUSSION THREAD

**Discussion by nerai** [paviad/GoSharp#36] [General]
My usual whitespace settings are completely different from what is mainly used in this codebase, so it is a bit cumbersome sometimes. I'm not sure how you feel about this @paviad, so I would like to ask for clarification what should be kept as is and what may be changed. I'm willing to adapt.

1. Should tabs be used for indentation (but _not_ formatting)?
I definitely think so, but I am not too religious about it.

2. Line breaks unix style (LF) or windows style (CRLF)?
I love LF, which is currently used. Some commonly used tools sometimes break with them, but it seems it was mostly fixed in VS 2017, so I'd suggest keeping it like this.

3. Space before opening parenthesis?
`foo (2)` instead of `foo(2)`
It is not currently applied. I find it easier to read.

4. Some braces on same line?
Currently all braces are on a new line. A more dense format (braces on new line only for classes and methods but not `if`, `for`, lambdas etc.) seems easier to read to me. Especially as there are _no_ long, complex lines in this project that would need space around them to be readable.

5. UTF8 BOM
UTF8 does not require a byte order marker. I suggest removing it.

Regardless of the decisions, it would be useful to have the relevant settings available in the repo so they can be imported into VS (and CodeMaid) easily. I can submit these for VS17. I'm currently abroad so I don't have VS13 and 15 with me.

**Comment by paviad** [comment]
I have a style I prefer, let me answer your questions:

1. I hate tabs with a passion - no tabs at all.
2. I did not pay any particular attention to line breaks, I'm using the defaults.
3. No space before parentheses
4. All opening braces on the same line, this has changed pretty recently for me as I used to prefer the opening brace on a new line, but recently I came to understand that to support source control and revisioning it is better if all blocks are surrounded by braces, even if they are a single line, and that the opening brace of that block should not be on a line of its own. This makes it so adding or removing lines from a block is shown more easily in diffs.

However having said all that, this code base was written before I came to this realization, so I have consciously not reformatted it, however I'm considering it.

5. I go with the defaults of VS2017 with regards to UTF8, so whatever it does is ok by me.

**Comment by paviad** [comment]
Checkout the branch `style-change` that I created with formatting changes. See if you have any comments about that.

**Comment by paviad** [comment]
I "upgraded" that branch to also include the latest C# features (7.2) - and renamed variables according to naming conventions. It's a huge cosmetic change, but one that I like.

**Comment by nerai** [comment]
Sorry, I have not yet had enough time to look closely. But your changes sound good.

**Comment by nerai** [comment]
Looking good. I agree with your reasoning of point 4.
